### PR TITLE
[terraform] flow log module

### PR DESCRIPTION
**Pull request by strcrzy**
to: @airbnb/streamalert-maintainers 
size: small
issue: #16 

# Stream Alert Flow Log Terraform Module
This Terraform module enables flow logs for specified VPCs, subnets, and ENIs.

## Components
Creates the following:
* A cloudwatch log group for the flow logs
* A subscription filter delivering the logs to the specified destination stream
* Enables flow logs for VPC resources specified in `var.targets`

### CODE REVIEW

File: terraform/modules/tf_stream_alert_flow_logs/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Stream Alert Flow Log Terraform Module
```
Comment (jacknagz) [inline terraform/modules/tf_stream_alert_flow_logs/README.md]: StreamAlert VPC Flow Log Terraform Module*

File: terraform/modules/tf_stream_alert_flow_logs/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+* StreamAlert: a serverless, real-time intrusion detection engine.
```
Comment (jacknagz) [inline terraform/modules/tf_stream_alert_flow_logs/README.md]: nix

File: terraform/modules/tf_stream_alert_flow_logs/main.tf [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  iam_role_arn    = "${aws_iam_role.flow_log_role.arn}"
+  traffic_type    = "ALL"
+}
+
+resource "aws_flow_log" "subnet_flow_log" {
+  count           = "${length(var.targets["subnets"])}"
+  subnet_id       = "${element(var.targets["subnets"],count.index)}"
+  log_group_name  = "${aws_cloudwatch_log_group.flow_log_group.name}"
+  iam_role_arn    = "${aws_iam_role.flow_log_role.arn}"
+  traffic_type    = "ALL"
```
Comment (jacknagz) [inline terraform/modules/tf_stream_alert_flow_logs/main.tf]: What are the traffic type options? Should we make it configurable?

**Comment by jacknagz** [comment]
Awesome work on this @strcrzy, left a couple comments.  Could you also run `terraform fmt` on the module?

File: terraform/modules/tf_stream_alert_flow_logs/main.tf (hunk 3 above)
Comment (strcrzy) [inline terraform/modules/tf_stream_alert_flow_logs/main.tf]: traffic type can be `ALL` `ACCEPT`, or `REJECT` -- i can definitely make this configurable

**Comment by jacknagz** [comment]
I'm thinking we might want to add support for this in the `variables.json` similar to how we do s3 events.  Thoughts?

**Comment by strcrzy** [comment]
will plug this in to `cluster_template` in the PR that updates it to use json

**Pull request merged**